[2024-11-28T22:22:35.277+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-11-28T22:22:35.289+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: brewery_ETL.transform_gold manual__2024-11-28T22:16:44.284541+00:00 [queued]>
[2024-11-28T22:22:35.295+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: brewery_ETL.transform_gold manual__2024-11-28T22:16:44.284541+00:00 [queued]>
[2024-11-28T22:22:35.296+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 2
[2024-11-28T22:22:35.305+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): transform_gold> on 2024-11-28 22:16:44.284541+00:00
[2024-11-28T22:22:35.311+0000] {standard_task_runner.py:72} INFO - Started process 1175 to run task
[2024-11-28T22:22:35.314+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'brewery_ETL', 'transform_gold', 'manual__2024-11-28T22:16:44.284541+00:00', '--job-id', '242', '--raw', '--subdir', 'DAGS_FOLDER/openbrewery_dag.py', '--cfg-path', '/tmp/tmpe80fw2it']
[2024-11-28T22:22:35.316+0000] {standard_task_runner.py:105} INFO - Job 242: Subtask transform_gold
[2024-11-28T22:22:35.360+0000] {task_command.py:467} INFO - Running <TaskInstance: brewery_ETL.transform_gold manual__2024-11-28T22:16:44.284541+00:00 [running]> on host dc991f0d715f
[2024-11-28T22:22:35.422+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='brewery_ETL' AIRFLOW_CTX_TASK_ID='transform_gold' AIRFLOW_CTX_EXECUTION_DATE='2024-11-28T22:16:44.284541+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-11-28T22:16:44.284541+00:00'
[2024-11-28T22:22:35.423+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-11-28T22:22:35.436+0000] {transform_gold.py:153} INFO - Iniciando a tarefa transform_gold
[2024-11-28T22:22:35.436+0000] {transform_gold.py:167} INFO - Data de execução: 2024-11-28
[2024-11-28T22:22:35.437+0000] {transform_gold.py:25} INFO - Iniciando criação da sessão Spark para transformação Gold
[2024-11-28T22:22:41.747+0000] {transform_gold.py:50} INFO - Sessão Spark criada com sucesso para transformação Gold
[2024-11-28T22:22:41.747+0000] {transform_gold.py:59} INFO - Lendo dados da camada Silver: s3a://breweris-etl-prod-sp/silver/date=2024-11-28/
[2024-11-28T22:22:55.565+0000] {transform_gold.py:91} INFO - Dados da camada Silver carregados: 50 registros
[2024-11-28T22:22:55.566+0000] {transform_gold.py:101} INFO - Iniciando agregação dos dados para camada Gold
[2024-11-28T22:22:55.566+0000] {transform_gold.py:104} INFO - Verificando as colunas brewery_type e localizacao
[2024-11-28T22:22:57.502+0000] {transform_gold.py:111} WARNING - Encontrados 50 registros com valores nulos em localizacao
[2024-11-28T22:22:57.503+0000] {transform_gold.py:114} INFO - Amostra dos dados antes da agregação:
[2024-11-28T22:22:58.591+0000] {logging_mixin.py:190} INFO - +------------+-----------+
|brewery_type|localizacao|
+------------+-----------+
|large       |null       |
|large       |null       |
|large       |null       |
|micro       |null       |
|micro       |null       |
+------------+-----------+
only showing top 5 rows
[2024-11-28T22:22:58.598+0000] {transform_gold.py:131} ERROR - Erro durante a agregação dos dados: local variable 'count' referenced before assignment
[2024-11-28T22:22:58.599+0000] {transform_gold.py:183} ERROR - Erro na tarefa transform_gold: local variable 'count' referenced before assignment
[2024-11-28T22:22:58.655+0000] {transform_gold.py:188} INFO - Sessão Spark finalizada para a tarefa transform_gold
[2024-11-28T22:22:58.656+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/opt/airflow/scripts/transform_gold.py", line 176, in transform_gold
    df_gold = aggregate_data(df_silver)
  File "/opt/airflow/scripts/transform_gold.py", line 119, in aggregate_data
    .agg(count("*").alias("quantity_of_breweries"))
UnboundLocalError: local variable 'count' referenced before assignment
[2024-11-28T22:22:58.673+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=brewery_ETL, task_id=transform_gold, run_id=manual__2024-11-28T22:16:44.284541+00:00, execution_date=20241128T221644, start_date=20241128T222235, end_date=20241128T222258
[2024-11-28T22:22:58.694+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-11-28T22:22:58.695+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 242 for task transform_gold (local variable 'count' referenced before assignment; 1175)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/opt/airflow/scripts/transform_gold.py", line 176, in transform_gold
    df_gold = aggregate_data(df_silver)
  File "/opt/airflow/scripts/transform_gold.py", line 119, in aggregate_data
    .agg(count("*").alias("quantity_of_breweries"))
UnboundLocalError: local variable 'count' referenced before assignment
[2024-11-28T22:22:58.714+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-11-28T22:22:58.744+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-11-28T22:22:58.746+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
